### DISCUSSION TITLE
classes/class_bool

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#175] [User-contributed notes]
# classes/class_bool

A built-in boolean type. Description: The bool is a built-in Variant type that may only store one of two values: true or false. You can imagine it as a switch that can be either turned on or off, o...

https://docs.godotengine.org/en/stable/classes/class_bool.html

**Comment by mathgeniuszach** [comment]
Interesting that > and < are implemented, but not >= or <=.

**Reply by Calinou** [reply]
I haven't checked the engine code, but I believe `>=` is internally implemented by the engine without needing to manually define an overload for it. The engine internally checks the overload for `>` then `==`, and evaluates to `true` if either of them are `true`. The same goes for `<=`.

**Reply by EdwardChanCH** [reply]
Opened a proposal: https://github.com/godotengine/godot-proposals/issues/14413
